src/testUtils/server.ts

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable require-atomic-updates */
+
+import type { FastifyTypedInstance } from '../services/fastify.js';
+import { makeServer } from '../services/server.js';
+
+export function setUpTestServer(): () => FastifyTypedInstance {
+  let server: FastifyTypedInstance;
+  beforeAll(async () => {
+    server = await makeServer();
+  });
+
+  beforeEach(async () => {
+    await server.close();
+    server = await makeServer();
+  });
+
+  afterEach(async () => {
+    await server.close();
```

Comment:
Have you checked what happens when `server.close()` is called multiple times?

I'm asking because, under certain circumstances, Fastify would internally call `server.close()` when you use `server.inject()` (like we do in our tests). For example, in the event of internal server errors.

Reply:
To clarify, it's a genuine question -- I don't know what happens when you call it twice. I'm just saying we should check what happens because we might have to call `server.close()` conditionally if it can only ever be called once.

Reply:
This does not seam to have any issue. I tested it and called close 2 times without any error

Reply:
Fantastic, let's ignore this then 🙌🏾